src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -410,6 +441,25 @@ public double getAlignAngle(){
             Math.atan2(VisionConstants.RED_SPEAKER_POSE.getY() - pose.getY(), VisionConstants.RED_SPEAKER_POSE.getX() - pose.getX()));
     }
 
+    /**
+     * Uses pigeon and rotational input to return a rotation that accounts for drift
+     * @return A rotation
+     */
+    public double headingControl(double rot, double xSpeed, double ySpeed){
+        if((!EqualsUtil.epsilonEquals(getAngularRate(0), 0, 3.5621085461121804E-4)&&EqualsUtil.epsilonEquals(Math.hypot(xSpeed, ySpeed),0,0.1))||!EqualsUtil.epsilonEquals(rot, 0, 3.5621085461121804E-4)){
```

Comment:
Where did you get some of these numbers (3.562...)?

Reply:
testing